internal/component/remote/aws/secrets_manager/secrets_manager.go

```diff
@@ -0,0 +1,194 @@
+package secrets_manager
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
+	"github.com/grafana/alloy/internal/component"
+	aws_common_config "github.com/grafana/alloy/internal/component/common/config/aws"
+	"github.com/grafana/alloy/internal/featuregate"
+	"github.com/grafana/alloy/syntax/alloytypes"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:      "remote.aws.secrets_manager",
+		Stability: featuregate.StabilityExperimental,
+		Args:      Arguments{},
+		Exports:   Exports{},
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+}
+
+type Component struct {
+	opts          component.Options
+	mut           sync.Mutex
+	health        component.Health
+	pollFrequency time.Duration
+	watcher       *watcher
+	updateChan    chan result
+	errors        prometheus.Counter
+	lastAccessed  prometheus.Gauge
+}
+
+var (
+	_ component.Component       = (*Component)(nil)
+	_ component.HealthComponent = (*Component)(nil)
+)
+
+type Arguments struct {
+	// Options allows the overriding of default AWS settings.
+	Options       aws_common_config.Client `alloy:"client,block,optional"`
+	SecretId      string                   `alloy:"id,attr"`
+	SecretVersion string                   `alloy:"version,attr,optional"`
+	PollFrequency time.Duration            `alloy:"poll_frequency,attr,optional"`
+}
+
+// DefaultArguments holds default settings for Arguments.
+var DefaultArguments = Arguments{
+	SecretVersion: "AWSCURRENT",
+}
+
+// SetToDefault implements syntax.Defaulter.
+func (a *Arguments) SetToDefault() {
+	*a = DefaultArguments
+}
```

Comment:
We've started to move away from the pattern of having default arguments as a variable because in some cases (like a default map or slice) memory gets accidentally shared and the defaults get overridden. 

```suggestion

// SetToDefault implements syntax.Defaulter.
func (a *Arguments) SetToDefault() {
	*a = Arguments{
		SecretVersion: "AWSCURRENT",	
	}
}
```